projects/js-toolkit/packages/js-toolkit-core/src/project/liferayCli/Deploy.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * SPDX-FileCopyrightText: © 2021 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ */
+
+import fs from 'fs';
+
+import FilePath from '../../file/FilePath';
+import LiferayJson from '../../schema/LiferayJson';
+import Project from './Project';
+
+export default class Deploy {
+	readonly dir: FilePath | null;
+
+	constructor(project: Project, liferayJson: LiferayJson) {
+		this._project = project;
+
+		if (
+			liferayJson.deploy === undefined ||
+			liferayJson.deploy.path === undefined
+		) {
+			this.dir = null;
+		}
+		else {
+			this.dir = new FilePath(liferayJson.deploy.path as string, {
+				posix: true,
+			}).resolve();
+		}
+	}
+
+	storeDir(deployDir: FilePath): void {
+		(this as object)['dir'] = deployDir.resolve();
+
+		const file = this._project.dir.join('.liferay.json');
+
+		const liferayJson = JSON.parse(
+			fs.readFileSync(file.asNative, 'utf8')
+		) as LiferayJson;
+
+		liferayJson.deploy = liferayJson.deploy || {};
+		liferayJson.deploy.path = this.dir.asPosix;
+
+		fs.writeFileSync(
```

Comment:
Mmm, not really, because it should point to `portal/bundles/osgi/deploy` or similar folders. So, they should exist right before the project is deployed.

:thinking: In fact, if it doesn't exist, it is probably better to fail the deployment so that the user realizes that the project is being deployed to a wrong place...